staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go

```diff
@@ -213,7 +215,48 @@ func (c *defaultAuthenticationInfoResolver) clientConfig(target string) (*rest.C
 	return setGlobalDefaults(&rest.Config{}), nil
 }
 
-func restConfigFromKubeconfig(configAuthInfo *clientcmdapi.AuthInfo) (*rest.Config, error) {
+func (c *defaultAuthenticationInfoResolver) execConfigFromAuthInfo(authInfoName string) runtime.Object {
+	if cluster := func() *clientcmdapi.Cluster {
+		// if we only have one, just use it
+		if len(c.kubeconfig.Clusters) == 1 {
+			for _, cluster := range c.kubeconfig.Clusters {
+				return cluster
+			}
+		}
+
+		// use the current context if possible
+		if len(c.kubeconfig.CurrentContext) > 0 {
```

Comment:
The client config used by the API server to authenticate to admission webhooks (set via `--admission-control-config-file`) could technically use an `exec` plugin since it is no different than setting any other config in the `user` section (the code already supports it, I just updated the wiring to pass in per-cluster `exec` config if set).

For example, in `master`, we check the current context to get the auth info:

https://github.com/kubernetes/kubernetes/blob/19d220073c911d9480d62062c27fedc826774492/staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go#L201-L210

Custom resource conversion webhooks appear to be anonymously contacted so this does not apply there.

I primarily did this for completeness since `exec` is supported in `master`:

https://github.com/kubernetes/kubernetes/blob/19d220073c911d9480d62062c27fedc826774492/staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go#L248-L250

Reply:
I am leaning towards dropping this change because the `exec` functionality is not [documented](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#authenticate-apiservers) and I doubt anyone really needs this level of flexibility here.  WDYT?

@mikedanese you seem to have forgotten to add a release note in #76919 or update the docs 😄

Reply:
Oh interesting that the authenticator webhook is (the only?) webhooks that use it. Maybe @liggitt can weigh in. I lean slightly towards leaving it out until someone requests that we add it, since we are primarily implementing this to improve the experience for kubeconfigs targeting the apiserver.

Reply:
I thought all of the webhooks loaded kubeconfig files normally:

https://github.com/kubernetes/kubernetes/blob/0072d8ae23878c8f01985a66dff27843169f691d/staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go#L75-L82

Reply:
> This place is "special" because the names of the user stanzas are meaningful (a single kubeconfig is used to select different user credentials based on the destination hostname)

Right, hence I do not really think the per-cluster config even makes sense here.  I will go ahead and drop it.